styles/theme-dark.less

```diff
@@ -57,6 +57,18 @@ there first. */
 
     // "Advanced" options background
     --advanced-bg: rgba(127, 81, 81, 0.5);
+
+    // Container colors and styles
+    --container-color-blue: blue;
+    --container-color-teal: teal;
+    --container-color-green: green;
+    --container-color-yellow: yellow;
+    --container-color-orange: orange;
+    --container-color-red: red;
+    --container-color-magenta: magenta;
+    --container-color-purple: purple;
+    --container-color-toolbar: #606060;
+    --container-indicator-shadow-style: inset -2px 0px 0px 0px;
```

Comment:
Since the shadow-style is all length and no color, it should probably go with the other measurements in `metrics-*.less`.

Also, it would probably look a little more consistent to use the same width as the active-tab marker--having it slightly narrower looks a little weird to my eye but maybe that's just me? In any case, what you want is something like: `calc(0px - var(--active-marker-bw))` (and yes, you need `0px` not `0`, I guess because one is a length and the other isn't...? CSS is weird 🙄)

Reply:
> Since the shadow-style is all length and no color, it should probably go with the other measurements

I was thinking that we might want to do stuff like make it fatter in light themes compared to dark themes where the colors stand out more by default. I can move it if you still want me to, though.

> having it slightly narrower looks a little weird to my eye but maybe that's just me?

I personally prefer having it thinner (with a dark theme, anyway), but I don't have very strong feelings here so I can make the change if you'd like.

My thinking is something like the active tab indicator needs to be visible and should pull attention. Information like what container a tab belongs to should be available, but it really doesn't need to draw focus. The container indicators are already pretty visible (especially bright ones like orange) with a dark theme. I'd also like to preserve as much available space as possible for the tab title.

That said, the indicators probably _could_ stand to be more visible with a light theme. What do you think about making that change, but only for light theme users?

Reply:
I noticed Firefox itself seems to use a 2px(ish) marker that's barely visible (at least to me). So maybe it's OK to do half-width.

One other idea I had: what about an underline, or an overline to mimic the style used in the browser UI itself? Something like:

```css
.folder-item[data-container-color] .text {
  text-decoration: overline var(--container-indicator-color) 1px;
}
```

<img width="275" alt="Screenshot 2022-03-05 105112" src="https://user-images.githubusercontent.com/1203893/156896394-216f4813-6347-4061-a726-8438379a0877.png">

<img width="266" alt="Screenshot 2022-03-05 105738" src="https://user-images.githubusercontent.com/1203893/156896567-ac4b965a-64bc-4d73-a9a2-6b948286fee8.png">

Not sure if that's over-correcting/too obtrusive, but wanted to throw it out there.

In any case, I think we could probably pick colors that work well with the theme and get to a consistent level of contrast.

Reply:
> One other idea I had: what about an underline, or an overline to mimic the style used in the browser UI itself? 

I think an overline would be too weird. Is there any interface stuff that uses overlines? I feel like I'd always end up having to double check which item something was applying to. Underlines as indicators are kind of ambiguous too, but at least an underline would be what you'd expect.

On the subjective side, and for sidebar tabs at least I don't like the underline instead of the side indicator. Partly because of the ambiguity of what item it's associated with, like I mentioned before and partly because I'm just used to the side indicator and it's how all the other vertical tabs addons I've used in the past worked. It could be an option to toggle between underline container indicators and the current style?

One thing I will say is that underlines would make a lot more sense in the actual Tab Stash Tab interface. Right now, with a big display, the indicator can be pretty far away from the title of the tab. Would it be too inconsistent to use the underline approach there and (possibly configurable) vertical indicators in the sidebar?

> I think we could probably pick colors that work well with the theme and get to a consistent level of contrast.

I think the current colors in the dark theme are pretty good. The only things I might change is the dark blue/green colors, which are probably a bit too dark. What looks good in a theme is really subjective though!

Do you want to pick the colors for the light theme indicators? I really have no preference at all there since I'm _never_ going to use it. :) Should probably always be variations on the default colors though, since it would be pretty weird to choose red in the container configuration and have it show up as blue or grey.

For the dark theme, I have more of an opinion on the colors but also feel free to tweak those as well if you want to!

Reply:
> On the subjective side, and for sidebar tabs at least I don't like the underline instead of the side indicator. Partly because of the ambiguity of what item it's associated with, like I mentioned before and partly because I'm just used to the side indicator and it's how all the other vertical tabs addons I've used in the past worked. It could be an option to toggle between underline container indicators and the current style?

Yeah, the more I think about it, the more I'm inclined to go with your original design, mainly for consistency reasons--it's similar to what others are doing and aligns more closely with Firefox's UI as well.

There's one other thing I want to bounce off of you though that's related to this PR--for the "Open Tabs" feature, I want to eventually add some visual indication of whether a tab is stashed or not. (The lack of such an indicator is part of why "Open Tabs" is still experimental.) I was originally leaning toward using that right-side space for such an indicator, but that's obviously not going to work now. Some ideas I had, but didn't like for various reasons:

* Italics (mixing fonts looks a bit weird)
* Underline (adds visual noise & not typically used this way; similar drawbacks as using it for containers)
* Change the text color somehow (becomes inconsistent with the text colors used for stashed tabs)
* Do something to the favicon like adding a shadow, border, filter, etc. (seems to cause performance problems when the hover state changes)

Only other thing I can think of is adding some kind of faint icon at the end of the title (a la the verified checkmark), but do you have any ideas?

> Do you want to pick the colors for the light theme indicators? I really have no preference at all there since I'm _never_ going to use it. :) Should probably always be variations on the default colors though, since it would be pretty weird to choose red in the container configuration and have it show up as blue or grey.
> 
> For the dark theme, I have more of an opinion on the colors but also feel free to tweak those as well if you want to!

I'll probably check what Firefox does and copy that. :)

Reply:
> the more I think about it, the more I'm inclined to go with your original design

Even I couldn't find a reason to complain about that!

> I want to eventually add some visual indication of whether a tab is stashed or not. 

I had that thought also and definitely agree it would be useful.

Sadly, I agree with the reasons for not liking the other ideas. I don't know about putting stuff on the title either, which could obscure it or make it more difficult to read. I already don't like how much space the "..." from truncation is taking up!

It seems like there's space for indicators on the left side, where the active tab indicator is. You could display, for example, a dot or something there. It could possibly superimposed over the active indicator (when it exists) so that active+stashed was still distinguishable. The only thing I'd suggest is use something pretty small and muted, it doesn't seem like a thing that should be calling for attention.

On a similar topic: a number of tab addons also have an indicator for new tabs that have been opened, but not yet viewed. I was thinking of using the left side for that also. Probably something like a thin grey vertical line, similar to the container indicator. This wouldn't conflict with the active indicator also, since of course as soon as a tab became active the unseen indicator would no longer apply.

> I'll probably check what Firefox does and copy that. :)

Sure. I believe the container definitions actually contain an exact color code in addition to the color name, by the way. 

However, since Tab Stash doesn't necessarily use the exact same background/interface colors as Firefox and also can react to the theme it's asked to use it seems like a curated set of container colors for light/dark themes could be nicer.

Reply:
Interestingly, Firefox itself (or at least, the official container extension) seems to use the same set of colors in both dark and light mode:

https://github.com/mozilla/multi-account-containers/blob/cdd8ebc66a7101f33c62de7ab9d8d744f280c6d4/src/css/popup.css#L211-L255

We can probably do the same here; see what you think about these?